torch_audiomentations/core/composition.py

```diff
@@ -0,0 +1,62 @@
+import random
+from typing import List
+
+import torch
+
+
+class Compose(torch.nn.Module):
+    """This class can apply a sequence of transforms to waveforms."""
+
+    def __init__(
+        self,
+        transforms: List[torch.nn.Module],
+        shuffle: bool = False,
+        p: float = 1.0,
+        p_mode="per_batch",
+    ):
+        """
+
+        :param transforms: List of waveform transform instances
+        :param shuffle: Should the order of transforms be shuffled?
+        :param p: The probability of applying the Compose to the given batch.
+        :param p_mode: Only "per_batch" is supported at the moment.
+        """
+        super().__init__()
+        self.transforms = transforms
+        self.p = p
+        if p_mode != "per_batch":
+            # TODO: Support per_example as well? And per_channel?
```

Comment:
I am not sure what would happen when `Compose` is `per_something` and sub-transforms are `per_something_else`.

Reply:
Let's say you have 8 audio snippets in a batch, and you want to apply the Compose to 50% of them. Then that would be `p=0.5, p_mode="per_example"` (which is not implemented yet). Each transform has its own mode and p_mode that is used independently from other transforms

Reply:
I agree that it doesn't make sense to add `mode=` in Compose

Reply:
I left it as a TODO for later: https://github.com/asteroid-team/torch-audiomentations/issues/34